src/scripts/main.js

```diff
@@ -2,10 +2,23 @@
 
 function waitFor(element, eventName) {
   // write your code here
+  return new Promise((resolve) => {
+    element.addEventListener(eventName, function handler() {
+      const message = `It was ${eventName} on the element: ${element.nodeName}, id: ${element.id}.`;
+
+      resolve(message);
+      element.removeEventListener(eventName, handler);
+    });
```

Comment:
This `waitFor` function is currently removing the event listener after the first time the event is triggered. Depending on your application's requirements, you might want to consider whether this is the desired behavior. If you want the function to keep listening for the event even after it has been triggered once, you should remove the line that removes the event listener.